forge/drivers/s3.js

```diff
@@ -131,9 +131,11 @@ module.exports = function (app) {
             }))
 
             let size = 0
-            objects.Contents.forEach(file => {
-                size += file.Size
-            })
+            if (objects.Contents) {
+                objects.Contents.forEach(file => {
```

Comment:
Isn't 'reduce()' better in this context? And isn't this missing tests?

Reply:
Possibly, I just added the gate round what was already there at this point. I will look at it when I add the tests.

Reply:
I'd be fine with stubbing most of it, just doing unit tests